### ISSUE TITLE
Spring boot 3

### ISSUE THREAD

**Issue by JeffAtDeere** [okta/okta-spring-boot#507]
:information_source: **If you have a question**, please post it on the [Okta Developer Forum](https://devforum.okta.com) instead. Issues in this repository are reserved for bug reports and feature requests only.

### I'm submitting a
  - [ ] bug report
  - [x] feature request

### Background info

Please upgrade this library to work with Spring boot 3. `WebSecurityConfigurerAdapter` was deprecated and was removed in Spring boot 3.

### Expected behavior

I should be able use okta-spring-boot with the current version of spring boot (i.e. 3).

### What went wrong?

Example fails to compile due to removed class `WebSecurityConfigurerAdapter`

### SDK Version
java 17
spring boot 3
okta-spring-boot 2.1.6

**Comment by arvindkrishnakumar-okta** [comment]
Thanks for posting! This will be addressed soon.On Dec 15, 2022, at 12:23 PM, JeffAtDeere ***@***.***> wrote:﻿

This message originated outside your organization.
  

ℹ️ If you have a question, please post it on the Okta Developer Forum instead. Issues in this repository are reserved for bug reports and feature requests only.
I'm submitting a

 bug report
 feature request

Background info
Please upgrade this library to work with Spring boot 3. WebSecurityConfigurerAdapter was deprecated and was removed in Spring boot 3.
Expected behavior
I should be able use okta-spring-boot with the current version of spring boot (i.e. 3).
What went wrong?
Example fails to compile due to removed class WebSecurityConfigurerAdapter
Steps to reproduce
SDK Version
java 17
spring boot 3
okta-spring-boot 2.1.6

**Comment by mraible** [comment]
In my experience, the Okta Spring Boot starter v2.1.6 works just fine with Spring Boot 3. I updated [this tutorial](https://developer.okta.com/blog/2022/06/17/simple-crud-react-and-spring-boot) to use it a couple of weeks ago. 

However, if you use [start.spring.io](http://start.spring.io/) via browser or cURL/HTTPie, it fails.

```
$ curl https://start.spring.io/starter.tgz \
  -d bootVersion=3.0.0 \
  -d javaVersion=17 \
  -d dependencies=web,data-rest,lombok,data-jpa,h2,okta \
  -d type=gradle-project \
  -d baseDir=resource-server
```

```json
{
  "timestamp": "2022-12-21T15:51:01.608+00:00",
  "status": 400,
  "error": "Bad Request",
  "message": "Dependency 'okta' is not compatible with Spring Boot 3.0.0",
  "path": "/starter.tgz"
}
```

@JeffAtDeere You can use a filter chain bean in place of the `WebSecurityConfigurerAdapter`. Our [Spring Boot docs](https://developer.okta.com/docs/guides/protect-your-api/springboot/main/#configure-required-authentication) show how to do this.

**Comment by arvindkrishnakumar-okta** [comment]
Just released 3.0.0 that supports Spring Boot 3.
